File: Data Structures & Algorithms/is-palindrome/submission-0.cpp
class Solution {
    bool isAlphanumeric(const char c) {
        return (c >= '0' && c <= '9') ||
               (c >= 'A' && c <= 'Z') ||
               (c >= 'a' && c <= 'z');
    }
public:
    bool isPalindrome(string s) {
        int left = 0, right = s.size() - 1;
        while (left < right) {
            while (left < s.size() - 1 && !isAlphanumeric(s[left])) {
                ++left;
            }
            while (right > 0 && !isAlphanumeric(s[right])) {
                --right;
            }
            if (left >= right) {
                return true;
            }
            if (tolower(s[left]) != tolower(s[right])) {
                return false;
            }
            ++left;
            --right;
        }        
        return true;
    }
};

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def registerLevel(self, levels: List[int], curr_level: int, root: Optional[TreeNode]):
        if not root:
            return
        if len(levels) == curr_level:
            levels.append(root.val)
        else:
            levels[curr_level] = root.val;
        self.registerLevel(levels, curr_level + 1, root.left)
        self.registerLevel(levels, curr_level + 1, root.right)
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        right_side_view = []
        self.registerLevel(right_side_view, 0, root)
        return right_side_view